src/zones/zonesNormalize.ts

```diff
@@ -29,26 +29,29 @@ export interface ZonesNormalizeOptions {
  * @param options - options
  * @returns array of zones
  */
-export function zonesNormalize(
-  zones: FromTo[] = [],
+export function zonesNormalize<Zone extends FromTo = FromTo>(
+  zones: Zone[] = [],
   options: ZonesNormalizeOptions = {},
-): FromTo[] {
+): Zone[] {
   const { exclusions = [] } = options;
   let { from = Number.NEGATIVE_INFINITY, to = Number.POSITIVE_INFINITY } =
     options;
 
   if (from > to) [from, to] = [to, from];
 
   zones = zones
-    .map((zone: FromTo) =>
-      zone.from > zone.to ? { from: zone.to, to: zone.from } : { ...zone },
+    .map((zone: Zone) =>
+      zone.from > zone.to
+        ? { ...zone, from: zone.to, to: zone.from }
+        : { ...zone },
     )
     .sort((a, b) => {
       if (a.from !== b.from) return a.from - b.from;
       return a.to - b.to;
     });
+
   if (zones.length === 0) {
-    zones.push({ from, to });
+    zones.push({ from, to } as Zone);
```

Comment:
This is an issue because we will let think that it is a Zone but it does not have the expected properties. It could yields to bugs and this can not be accepted.

@targos I guess we could make function overload ? But I'm not sure this would really help

```ts
// Overload for when the array is empty
function test(array: []): FromTo;

// Overload for when the array is not empty
function test(array: number[]): Zone;

function test(array: number[]): FromTo | Zone {}
```

Otherwise we need to create a new method that just throws if the number of zones are empty

Reply:
Empty array does not exist in TypeScript